project/FrontEnd/collaborative_science_platform/lib/screens/graph_page/widgets/graph_node_popup.dart

```diff
@@ -89,11 +91,14 @@ class _NodeDetailsPopupState extends State<NodeDetailsPopup> {
             crossAxisAlignment: CrossAxisAlignment.start,
             mainAxisSize: MainAxisSize.min,
             children: [
-              AnnotationText('Title: ${node.nodeTitle}'),
+              AnnotationText('Title: ${utf8.decode(node.nodeTitle.codeUnits)}'),
               Text(
                   'Contributors: ${node.contributors.map((user) => "${user.firstName} ${user.lastName} (${user.email})").join(", ")}'),
               Text('Publish Date: ${getDurationFromNow(node.publishDate!)}'),
-              AnnotationText('Theorem Content: ${node.theorem?.theoremContent ?? "No theorem"}'),
+              TeXView(
+                  renderingEngine: TeXViewRenderingEngine.katex(),
+                  child: TeXViewDocument(
+                      '<b>Theorem Content:</b> ${utf8.decode(node.theorem!.theoremContent.codeUnits) ?? "No theorem"}')),
```

Comment:
Also this part got my attention. You are using "!" operator which is telling this value can't be null.
So you are claiming theorem can't be null. However it indeed can be. You need to do something like this.

node.theorem == null ? "No theorem" : node.theorem.theoremContent.codeUnits


I'll fix this along conflicts

Reply:
I resolved this